share/shwap/p2p/bitswap/block.go

```diff
@@ -0,0 +1,38 @@
+package bitswap
+
+import (
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	logger "github.com/ipfs/go-log/v2"
+
+	"github.com/celestiaorg/rsmt2d"
+
+	"github.com/celestiaorg/celestia-node/share"
+)
+
+var log = logger.Logger("shwap/bitswap")
+
+// PopulateFn is a closure produced by a Block that validates given
+// serialized Shwap container and populates the Block with it on success.
+type PopulateFn func([]byte) error
+
+// Block represents Bitswap compatible Shwap container.
+// All Shwap containers must have a RegisterBlock-ed wrapper
+// implementing the interface to be compatible with Bitswap.
+type Block interface {
+	// String returns string representation of the Block
+	// to be used as map key. Might not be human-readable.
```

Comment:
> Might not be human-readable.

Should we make it human-readable? We use hex encoding extensively across Celestia projects. Hm, is `not be human-readable` refers to hex-encoded string?

Reply:
> Hm, is not be human-readable refers to hex-encoded string?

It refers to simple byte to string conversion with `string()`

Reply:
So, maybe we should prefer hex-encoded bytes, wdyt?

Reply:
Looks like this method is dropped, resolving.